libraries/abstractions/platform/freertos/iot_network_udp.c

```diff
@@ -0,0 +1,544 @@
+
+        status = IOT_NETWORK_SYSTEM_ERROR;
+    }
+
+    IOT_FUNCTION_EXIT_NO_CLEANUP();
+}
+/*-----------------------------------------------------------*/
+
+size_t IotNetworkUdp_SendTo( void * pConnection,
+                           const uint8_t * pMessage,
+                           size_t messageLength )
+{
+    size_t bytesSent = 0;
+    _networkConnection_t * pNetworkConnection = ( _networkConnection_t * ) pConnection;
+
+    if( pConnection == NULL )
+    {
+        IotLogWarn( "Invalid connection parameter" );
+    }
+    else
+    {
+        bytesSent = SOCKETS_SendTo( pNetworkConnection->socket,
+                                    ( void * ) pMessage,
+                                    messageLength,
+                                    0,
+                                    &pNetworkConnection->serverAddr,
+                                    sizeof( SocketsSockaddr_t ) );
+    }
+
+    return bytesSent;
+}
+/*-----------------------------------------------------------*/
+
+size_t IotNetworkUdp_ReceiveFrom( void * pConnection,
+                              uint8_t * pBuffer,
+                              size_t bytesRequested )
+{
+    size_t bytesReceived = 0;
+    int32_t socketStatus = 0;
+    _networkConnection_t * pNetworkConnection = ( _networkConnection_t * ) pConnection;
+    SocketsSockaddr_t serverAddress = { 0 };
+    size_t addrLength = 0;
+
+    if( pConnection == NULL )
+    {
+        IotLogWarn( "Invalid connection parameter" );
+    }
+    /* if Receive was called as a result of asynch callback, check if receive had failed.*/
+    else if( ( xTaskGetCurrentTaskHandle() == pNetworkConnection->receiveTask ) && ( pNetworkConnection->bufferedByteValid == false ) )
+    {
+        IotLogWarn( " Async receive indicated failure\n" );
+    }
+    else
+    {
+        socketStatus = SOCKETS_RecvFrom( pNetworkConnection->socket,
+                                         pBuffer + bytesReceived,
+                                         bytesRequested,
+                                         0,
+                                         &serverAddress,
+                                         &addrLength );
+
+        if( socketStatus <= 0 )
+        {
+            IotLogError( "Failed to receive requested bytes: %d, received: %d socketStatus: %d", bytesRequested, bytesReceived,
+                         socketStatus );
+        }
+
+        bytesReceived += socketStatus;
+        configASSERT( bytesReceived == bytesRequested );
+    }
+
+    if( ( bytesReceived > 0 ) && ( pNetworkConnection->strictServerAddrCheck ) )
+    {
+        /* if strict server address check is enabled, only accept packets from the server that was used for Send */
+        if( serverAddress.ulAddress != pNetworkConnection->serverAddr.ulAddress )
+        {
+            IotLogError( "Server address does not match and strict server address check enabled." );
+            bytesReceived = 0;
```

Comment:
Although ```bytesReceived``` is set to ```0```, but ```pBuffer``` has the data in it. Should it be cleared off as well?